src/flight_control/av_state.cpp

```diff
@@ -114,20 +113,23 @@ State AvState::fromThrustSequence(DataDump dump)
     {
         return State::ERRORFLIGHT;
     }
-    // If the engine is properly ignited and a liftoff has been detected we go to LIFTOFF state
-    // TODO: ensure those are the right checks
-    else if (dump.prop.fuel_inj_pressure >= IGNITER_PRESSURE_WANTED && dump.nav.speed.z > SPEED_ZERO && dump.nav.altitude > ALTITUDE_ZERO)
-    {
-        return State::LIFTOFF;
-    }
+
     // If the pression is too low in the igniter or combustion chamber we go to the ARMED state
     // a bit agressive TODO: have  a counter or a sleep
     //TODO: check FAILEDIGNIT
-    else if (dump.prop.igniter_pressure < IGNITER_PRESSURE_WANTED || dump.prop.chamber_pressure < CHAMBER_PRESSURE_WANTED)
+    else if (dump.prop.igniter_pressure < IGNITER_PRESSURE_WANTED || dump.prop.chamber_pressure < CHAMBER_PRESSURE_WANTED || dump.prop.fuel_inj_pressure < INJECTOR_PRESSURE_WANTED_MIN)
     {
         return State::ARMED;
     }
```

Comment:
we should not have sensor value check to go back to armed but a msg from DPR/PRB must create an event "ignition failed" and once detected must rewrite false to it